### PR TITLE
Add File::Stat test

### PR DESCRIPTION
**Pull request by snaka**
There were a few things that I found annoying, so it would be helpful if you could check them out just in case.

(I also wondered about the following, though it is a minor point)

I ran `bin/annotate-with-rdoc core/file.rbs`, but no annotations were added.
I don't know whether this is the correct behavior, or there is something wrong with my environment or operation.

### CODE REVIEW

File: core/file.rbs [hunk 1]
```diff
@@ -1042,8 +1042,6 @@ class File::Stat < Object
 
   def rdev_minor: () -> Integer
 
-  def read: (?int? length, ?string outbuf) -> String?
-
```
Comment (snaka) [inline core/file.rbs:1046]: `File::Stat#read` does not seem to exist.

I removed the definition of this method as it seemed unneeded.

It seems to have been inherited from the Sorbet RBI file from which this file originated.

https://github.com/sorbet/sorbet/blob/a0c88fe9ce2ce713a1e0b386460704284d50c568/rbi/core/file.rbi#L1822-L1829

File: core/file.rbs [hunk 2]
```diff
@@ -984,7 +984,7 @@ File::Constants::WRONLY: Integer
 class File::Stat < Object
   include Comparable
 
-  def initialize: (String file) -> Object
```
Comment (snaka) [inline core/file.rbs:987]: As far as I can see from the existing RBS file, the return type of the `initialize` method is `void` in some places.

I believe that the return value of a constructor can only be an instance of the class itself.

Therefore, the return value of the constructor should be `self`.

What are the cases in which it is appropriate to use `void`?

File: test/stdlib/File_Stat_test.rb [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+  def test_writable_real?
+    assert_send_type "() -> bool",
+                      File::Stat.new(__FILE__), :writable_real?
+  end
+
+  def test_zero?
+    assert_send_type "() -> bool",
+                      File::Stat.new(__FILE__), :zero?
+  end
+end
```
Comment (snaka) [inline test/stdlib/File_Stat_test.rb:235]: I ran `bin/annotate-with-rdoc core/file.rbs`, but no annotations were added.

I don't know whether this is the correct behavior, or whether there is something wrong with my environment or my operation.

File: core/file.rbs (hunk 2 above)
Comment (snaka) [reply core/file.rbs:987]: Thanks for your comment.
Now I know the meaning of `new`, `allocate` and `initialize` in Ruby.

**Review by soutaro — APPROVED** [review]
Thank you! 👏

**Comment by snaka** [comment]
Linked
https://github.com/ruby/rbs/issues/1

**Pull request merged**